python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -604,12 +660,18 @@ def _kill_all_running_actors(self):
 
     def _kill_running_actor(self, actor: ray.actor.ActorHandle):
         """Kill the provided actor and remove it from the pool."""
+        pg = self._actor_to_placement_groups.pop(actor, None)
+        if pg:
+            ray.util.remove_placement_group(pg)
```

Comment:
I don't think we should remove the PG here. Because other bundles in the PG can be still used.  also I think it's a bit hacky to add special logic for PGs here. 
I have 2 alternative solutions in mind:
* let the application handles PG removal after execution finishes. The drawback is scaling down will be delayed a bit. 
* instead of exposing `ray_remote_args_fn`, expose something lower level, such as "on_creating_actor", "on_actor_killed". 
I slightly prefer (1). But I think vLLM should eventually move away from PGs,  adding a temporary internal hook is acceptable as well. 
cc @c21

Reply:
> let the application handles PG removal after execution finishes. The drawback is scaling down will be delayed a bit.

Got it, +1 for option 1. The user-defined function can memorize the created PGs into a list, and remove all PGs at end of user program. WDYT?